### PR TITLE
Implement touch event support in MediaControl

### LINKED ISSUE #1205: iOS native controls appear when exiting full screen
**Browser**: chrome

**OS**: iOS 10.1.1 iPhone

**Clappr Version**: 0.2.64

**Steps to reproduce**:

- clIck play
- player plays full screen
- click full screen button to exit full screen
- playback continues inline (new iOS feature)
- native iOS control bar now appears below media control.

Not sure if this is something we can fix or an iOS bug.

https://goo.gl/photos/MebkTzzftNr8BZSE7

### PR DESCRIPTION
**Pull request by bikegriffith**
Improves usability on mobile devices by allowing finger-based dragging of the scrubber.

This PR also fixes #1205 by explicitly removing the `controls` attribute after `video. webkitendfullscreen` is fired.

### CODE REVIEW

File: src/components/media_control/media_control.js [hunk 1]
```diff
@@ -110,26 +125,36 @@ export default class MediaControl extends UIObject {
...
-      this.listenTo(this.container, Events.CONTAINER_VOLUME, this.onVolumeChanged)
-      if (this.container.playback.el.nodeName.toLowerCase() === 'video') {
-        // wait until the metadata has loaded and then check if fullscreen on video tag is supported
-        this.listenToOnce(this.container, Events.CONTAINER_LOADEDMETADATA, this.onLoadedMetadataOnVideoTag)
-      }
+    if (!this.container) {
+      return;
+    }
+
+    Mediator.on(`${this.options.playerId}:${Events.PLAYER_RESIZE}`, this.playerResize, this)
```
Comment (leandromoreira) [inline src/components/media_control/media_control.js]: I think, maybe we should go first as a container event or core =/ but I dunno, thoughts @clappr/core ?

File: src/components/media_control/media_control.js [hunk 2]
```diff
@@ -110,26 +125,36 @@ export default class MediaControl extends UIObject {
...
+    this.listenTo(this.container, Events.CONTAINER_ENDED, this.ended)
+    this.listenTo(this.container, Events.CONTAINER_VOLUME, this.onVolumeChanged)
+
+    var video = this.container.playback.el;
+    if (video.nodeName.toLowerCase() === 'video') {
+      // wait until the metadata has loaded and then check if fullscreen on video tag is supported
+      this.listenToOnce(this.container, Events.CONTAINER_LOADEDMETADATA, this.onLoadedMetadataOnVideoTag)
+
+      // watch for user to come out of fullscreen and remove the native controls that iOS tends to add
+      video.addEventListener('webkitendfullscreen', function() {
```
Comment (leandromoreira) [inline src/components/media_control/media_control.js]: I like this but we should handle this in a different manner, @towerz knows better.

Comment (kslimani) [inline src/components/media_control/media_control.js]: Maybe use `Events.CORE_FULLSCREEN` event boolean value ?

Comment (towerz) [inline src/components/media_control/media_control.js]: Indeed, @kslimani is right. There's already an event for fullscreen status change.

File: src/components/media_control/media_control.js (hunk 1 above)
Comment (towerz) [inline src/components/media_control/media_control.js]: About the PLAYER_RESIZE event, perhaps we should change the `triggerResize` event on the core to trigger this event as a core event as well. This way we won't need to rely on the Mediator for events that are internal to the player instance.

File: src/components/media_control/media_control.js (hunk 2 above)
Comment (bikegriffith) [inline src/components/media_control/media_control.js]: `CORE_FULLSCREEN` doesn't work on iOS when you're using the native controls to dismiss.  This is the only way I could find to let it work.

File: src/components/media_control/media_control.js (hunk 1 above)
Comment (bikegriffith) [inline src/components/media_control/media_control.js]: FWIW, I didn't actually change any of this.  I just made the `if` block return above to remove a level of indentation.

File: src/components/media_control/media_control.js (hunk 2 above)
Comment (kslimani) [inline src/components/media_control/media_control.js]: Good point. On iOS the event must be bind on the video element itself. I think (_but i did not test yet)_ that it should be possible to do late event binding somewhere in core if resolved playback container is a video element. @bikegriffith i will take a look this week to attempt to propose a fix for having CORE_FULLSCREEN trigger on iOS.

Comment (towerz) [inline src/components/media_control/media_control.js]: @bikegriffith There's a problem with this implementation. Perhaps it should be implemented on the HTML5Video playback itself. There's an option for the playback that tells it to use the default video tag controls (`playback.controls`), and this code would not follow the configuration correctly.

**Comment by bikegriffith** [comment]
Shutting this down.  The implementation in this PR didn't really work on Android.  Not sure how to easily bend what we have in our fork to work here.

**Pull request closed without merging**